docs/pages/guides/monorepos.md

```diff
@@ -92,10 +93,79 @@ config.resolver.nodeModulesPaths = [
+
+As your monorepo increases in size, watching all files within the monorepo becomes slower. If you want to speed that up, you can (dynamically) watch the relevant paths to minimize the watchers. E.g.
+
+```js
+const path = require('path');
+const workspaceRoot = path.resolve(__dirname, '../..');
+
+// If your monorepo tooling can give you a list of monorepo dependencies from a workspace,
+// you can automate this instead of hardcoding the paths.
+const appMonorepoPackages = ['ui', 'eslint-config'];
+const appPackagesPaths = appMonorepoPackages.map(
+  // Get the path of the package folder, not just the entry point.
+  packageName => path.resolve(workspaceRoot, 'packages', packageName)
+);
+
+// 1. Watch the local app folder, and all related packages (limiting the watchers and speeding it up)
+config.watchFolders = [__dirname, ...appPackagesPaths];
+```
+
+</Collapsible>
+
+<Collapsible summary="2. Why do we need tell Metro how to resolve packages?">
+
+This option is important to resolve libraries in the correct **node_modules** directories. Monorepo tooling, like Yarn, usually creates two different **node_modules** directories which are used for a single workspace.
+
+1. **apps/mobile/node_modules** - The "project" folder
+2. **node_modules** - The "root" folder
+
+Yarn uses the root folder to install packages that are used in multiple workspaces. If a workspace uses a different package version, it installs that different version in the project folder.
+
+We have to tell Metro to look in these two folders. The order is important here because the project folder **node_modules** can contain specific versions we use for our app. When the package does not exist in the project folder, it should try the shared root folder.
+
+</Collapsible>
+
+<Collapsible summary="3. Why do we need to disable the hierarchical lookup?">
+
+This option is important for certain edge cases, such as a monorepo that includes multiple versions of the `react` package. Let's say you have the following monorepo:
+
+1. **apps/marketing** - A simple Next.js website to attract new users. (uses `react@17.x.x`)
+2. **apps/mobile** - Your awesome Expo app. (uses `react@18.x.x`)
+3. **apps/web** - Your awesome Next.js website. (uses `react@17.x.x`)
+
+With monorepo tooling like Yarn, React is installed in two different **node_modules** folders.
+1. **node_modules** - The root folder, contains `react@17.x.x`.
+2. **apps/mobile/node_modules** - The Expo app's folder, contains `react@18.x.x`. 
+
+React Native libraries, including Expo modules, usually don't add `react@18.x.x` as peer dependency. As a result, monorepo tooling, like Yarn, will install these dependencies to the root **node_modules** directory, for example:
```

Comment:
This might read more smoothly. Also, if I understand correctly, the issue is that packages don't specify react in their peer deps, not that they don't specify react@18 (in other words, if they specified react@* that would actually fix this issue):
```suggestion
Expo modules and React Native libraries usually don't add `react` as a peer dependency. As a result, monorepo tooling, like Yarn, will install these dependencies to the root **node_modules** directory, for example:
```